### PR TITLE
[Aftershock] Skull gun

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary
```SUMMARY: Mods "[Aftershock] Skullgun"```

#### Purpose of change
This PR is #40635 with the supporting C++. please refer to that PR for details. Additionally this pr obsoletes that one, due to this pr having the attached json

#### Describe the solution
Adds a new spell flag: NO_FAIL which makes the spell never fail when casting it.
Adds the ability to have CBM activation cast a spell.

#### Testing
spawned in the skullgun and installed it, and checked that activating it casts the "snapback" spell, which works. A note, i found that the bionic shotgun won't be able to be fired as it has no bullets, which i had to check because this doesn't either. which means this is a more general bug that needs to be fixed separately and is out of scope.

### CODE REVIEW

**Comment by nexusmrsep** [comment]
Unless something broke it, the bionic shotgun should be reloadable with any 12g ammo. I've used it for testing few days ago and it worked.

File: data/mods/Aftershock/player/bionics.json [hunk 1]
```diff
@@ -57,5 +57,18 @@
     "description": "You've worked for some nasty people.  People who installed a bomb at the top of your spine.  They are all dead now but there is unfortunately a dead man switch if you don't check in roughly every thirty days.  You need this out and fast.",
     "occupied_bodyparts": [ [ "head", 2 ] ],
     "enchantments": [ "cranial_explosion" ]
+  },
+  {
+    "type": "bionic",
+    "name": { "str": "Skullgun CBM" },
```
Comment (ZhilkinSerg) [inline data/mods/Aftershock/player/bionics.json:63]: `id` is missing

Comment (KorGgenT) [reply data/mods/Aftershock/player/bionics.json:64]: fixed

**Comment by Ilysen** [comment]
Popping in from greener pastures to say that this bionic and its flavor is extremely dumb and I love it for that. That is all.

**Pull request merged**